### PR TITLE
fix(curriculum) more comprehensive test for 24 game

### LINKED ISSUE #47028: 24 Game is missing answers.
### Describe the Issue

There are tons of different ways to write this algorithim, that will come up with tons of different variations of the acceptable equation that reaches 24. 
My algorithim came up with the following solutions:
for set 4878 my algorithm returns 7 * 8 - 4 * 8 which is 24, but that will fail the test
for set (1234) my algorithm returns ( 3 + ( 1 + 2 ) ) * 4 which is 24, but will fail the test. (for set (1234)
and for set 6897 we get  6 * 8 / ( 9 - 7 ) which is 24, but will fail the test
and  3 * 4 / 1 * 2 and i'm sure plenty more...

The question does not specify a specific equation it wants, only that it should be 24 and fit the format. So for this reason, I think its broken, and either needs its answer bank revised, or needs to have numbers with only 1 or 2 solutions chosen for the tests.

### Affected Page

https://www.freecodecamp.org/learn/coding-interview-prep/rosetta-code/24-game

### Your code

```
function solve24 (numStr) {
  let nums = numStr.split("").map((elem)=>{
    return [elem, ""+elem];
  })
  let answer = false;
  let ops = ['+','-','*','/']

  const recursive24 = (nums) => {
    if(nums.length == 1){
      if (nums[0][0] == 24){
        answer = nums[0][1];
      }
      return
    }
    if(answer){
      return
    }
    // select first number
    for(let x = 0; x < nums.length; x++){
      //select a operator
      for(let j = 0; j < ops.length; j++){
          //select second number
        for(let k = 0; k < nums.length; k++){
          if (k == x){
            continue
          }
          // copy new array, remove nums from it.
          let newArr= nums.slice().filter((val, index)=>{
            if (index ==k || index == x){
              return false
            }
            return true
          });
          let exp = nums[x][1]+ops[j]+nums[k][1];
          if (j<=1 && newArr.length > 0){
            exp="("+exp+")";
          }
          let val = eval(exp)
          newArr.push([val,exp])
          recursive24(newArr)

        }
      }
    }
  }
  recursive24(nums);
  console.log(answer)
  if(answer===false){
    return "no solution exists"
  }
  return answer;
}
```

### Expected behavior

Should pass the tests, but it fails because the answer bank does not have enough equation results.

### Screenshots

_No response_

### System

Desktop
free code camp terminal

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jeremylt**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

I still need to test this, but I think this will finally fix the 24 Game challenge for good. This PR adds a small string parsing script to check the user output and see if it evaluates correctly to 24.

Closes #47028

### CODE REVIEW

File: curriculum/challenges/english/10-coding-interview-prep/rosetta-code/24-game.md [hunk 1]
```diff
@@ -39,85 +39,118 @@ Implement a function that takes a string of four digits as its argument, with ea
...
+      case "-":
+      case "*":
+      case "/":
+        while (stack.length &&
+               PRECIDENCE_[c] < PRECIDENCE_[stack[stack.length-1]]) {
+          postfix += stack.pop();
+        }
+        stack.push(c);
+        break;
+      default:
```
Comment (jeremylt) [inline curriculum/challenges/english/10-coding-interview-prep/rosetta-code/24-game.md:122]: I could perhaps explicitly error on a non-digit here to make this code more robust

**Comment by jeremylt** [comment]
Ok, should be good to review and test now - it worked for me in GitPod

**Review by naomi-lgbt — APPROVED** [review]
Looks good to me!

**Review by gikf — APPROVED** [review]
Looks good! :+1:

**Pull request merged**